### ISSUE TITLE
wise loss function

### ISSUE THREAD

**Issue by cryax** [yoyololicon/pytorch-wise-ale#1]
Hi,

When compared to the loss function (equation 21) in the paper and your loss function WiSE_UB, why do we need to compute logAB_max here?

**Comment by yoyolicoris** [comment]
Good question.
This is a [LogSumExp](https://en.wikipedia.org/wiki/LogSumExp) trick to prevent numerical overflow.
In my early implementation, I found the inner product of AB in equation 21 would easily exploded when A is small. The original author avoid it by keeping A not smaller than a given threshold, but after some emails and discussions, we agree it's the best practice to do the operation in log domain.
